### PR TITLE
Add conda dependencies and testing to all vcflib tools

### PR DESCRIPTION
**Pull request by bgruening**
Depends on https://github.com/bioconda/bioconda-recipes/pull/1619 and a new package is needed with the new rc1 version of VCFtools.

### CODE REVIEW

**Comment by bgruening** [comment]
This one migrates the `vcf_tools` to conda dependencies and introduces parallel-travis testing, which should speed up testing dramatically for such PRs.
These tools can still be improved for styling and to stick more to our guideline but I'm running out of time.

Overall I think this PR is ready to go. Please review :)

**Comment by bgruening** [comment]
Can we merge this @galaxyproject/iuc ?

**Comment by gregvonkuster** [comment]
Fine with me, but unfortunately I feel that I am falling further and further behind on how this new conda stuff is working.  I'm trying to keep up, but I think I was behind from the start, so never completely got the context I needed initially to really understand it.  I'm sure there is documentation I've missed.  Can you point me to it?  ;)

File: tool_collections/vcflib/vcfcheck/vcfcheck.xml [hunk 1]
```diff
@@ -19,27 +19,27 @@
...
-	   <!-- Validators are commented to allow users apply too to any build. May need to be revised in the future
-		<validator type="unspecified_build" />
-		<validator type="dataset_metadata_in_data_table" table_name="fasta_indexes" metadata_name="dbkey" metadata_column="1" message="Sequences are not currently available for the specified build." />
-	   -->
-	 </param>
+            <param name="ref_file" type="select" label="Select reference genome">
+               <options from_data_table="fasta_indexes">
+                 <!--<filter type="data_meta" key="dbkey" ref="input_bam" column="value"/>-->
+               </options>
+                <validator type="no_options" message="A built-in reference genome is not available for the build associated with the selected input file"/>
```
Comment (nsoranzo) [inline tool_collections/vcflib/vcfcheck/vcfcheck.xml]: Indentation. This file should probably completely re-written using a coherent indentation.

File: tool_collections/vcflib/vcfrandomsample/vcfrandomsample.xml [hunk 2]
```diff
@@ -1,55 +1,55 @@
...
+    <inputs>
+        <param format="vcf" name="vcf_input" type="data" label="Select random sites from"/>
+        <param name="rate" type="float" value="0.1" label="base sampling probability per locus" help="-r, --rate"/>
+        <param name="random_seed" type="integer" value="17823618326" label="use this random seed"
+            help="-p, --random-seed; use this to make results reproducible" />
+        <conditional name="adv_options">
+            <param name="adv_options_selector" type="select" label="Advanced controls" help="Allows you to specify options that are not listed above">
+                <option value="no">Don't use advanced options</option>
+                <option value="use_adv_controls">Use advanced options</option>
+          </param>
```
Comment (nsoranzo) [inline tool_collections/vcflib/vcfrandomsample/vcfrandomsample.xml]: Indentation.

File: tool_collections/vcflib/vcfvcfintersect/vcfvcfintersect.xml [hunk 3]
```diff
@@ -1,103 +1,109 @@
...
        ln -s "${reference_source.ref_file}" "${reference_fasta_filename}" &amp;&amp;
     #else:
        #set $reference_fasta_filename = str( $reference_source.ref_file.fields.path )
     #end if
-    
-    #if $adv_options.adv_options_selector == True:
-    
-    vcfintersect "${adv_options.adv_options_input}" ${loci} ${invert} -r "${reference_fasta_filename}" -w "${window_size}" ${isect_union} "${vcf_input1}" "${vcf_input2}" > "${out_file1}"
 
+    #if $adv_options.adv_options_selector == True:
```
Comment (nsoranzo) [inline tool_collections/vcflib/vcfvcfintersect/vcfvcfintersect.xml]: s/True/"use_adv_controls"/

File: tool_collections/vcflib/vcfcheck/vcfcheck.xml (hunk 1 above)
Comment (bgruening) [inline tool_collections/vcflib/vcfcheck/vcfcheck.xml]: This was not the aim of this PR. But if you like I can go over all these files.

File: tool_collections/vcflib/vcfrandomsample/vcfrandomsample.xml [hunk 4]
```diff
@@ -1,55 +1,55 @@
...
+    <expand macro="requirements"/>
+    <expand macro="stdio" />
+    <command>
+        vcfrandomsample
+            #if $adv_options.adv_options_selector == "use_adv_controls":
+                -s "${adv_options.scale_by}"
+            #end if
+            -p ${random_seed}
+            -r ${rate}
+           "${vcf_input}" > "${out_file1}"
```
Comment (nsoranzo) [inline tool_collections/vcflib/vcfrandomsample/vcfrandomsample.xml]: Indentation

File: tool_collections/vcflib/vcfrandomsample/vcfrandomsample.xml [hunk 5]
```diff
@@ -1,55 +1,55 @@
...
+        <param name="random_seed" type="integer" value="17823618326" label="use this random seed"
+            help="-p, --random-seed; use this to make results reproducible" />
+        <conditional name="adv_options">
+            <param name="adv_options_selector" type="select" label="Advanced controls" help="Allows you to specify options that are not listed above">
+                <option value="no">Don't use advanced options</option>
+                <option value="use_adv_controls">Use advanced options</option>
+          </param>
+          <when value="" />
+          <when value="use_adv_controls">
+                <param name="scale_by" type="text" value="AF" label="scale sampling likelihood by this INFO field" help="-s, --scale-by; this only works for Float info fields such as AF" />
```
Comment (nsoranzo) [inline tool_collections/vcflib/vcfrandomsample/vcfrandomsample.xml]: Indentation.

**Comment by nsoranzo** [comment]
Thanks @bgruening, I have added some dos2unix and indentation fixes, merging now.

**Pull request merged**